Improved: List and Grid (OFBIZ-11345)

PierreSmits opened the pull request:
According to the definition in widget-form.xsd the use of a combination of a form with type="list" is deprecated in favour of a grid.
Refactor various list forms into grids.
Refactor various list form references in screens.

modified in product component:
ConfigScreens.xml: from form ref to grid ref , additional cleanup
ConfigForms.xml: from form definition with list ref to grid definition with list ref, additional clean-up

PierreSmits committed:
Improved: List and Grid (OFBIZ-11345)

According to the definition in widget-form.xsd the use of a combination of a form with type="list" is deprecated in favour of a grid.
Refactor various list forms into grids.
Refactor various list form references in screens.

modified in product component:
ConfigScreens.xml: from form ref to grid ref , additional cleanup
ConfigForms.xml: from form definition with list ref to grid definition with list ref, additional clean-up

JacquesLeRoux commented:
Hi Pierre,

Is it normal that ProductConfigOptionList and ListProductConfigItem are transformed to grid w/o calls to them?

PierreSmits commented:
Good morning Jacques, 

You mean like there are no triggers (menu-items) and request- and view-maps to show screens and data for these two? or other code elements referencing the screen?

JacquesLeRoux commented:
It's about no calls for them in ConfigScreens.xml

PierreSmits committed:
Improved: Grid and List (OFBIZ-11345)

According to the definition in widget-form.xsd the use of a combination of a form with type="list" is deprecated in favour of a grid.
Refactor various list forms into grids.
Refactor various list form references in screens.

Overlooked in ConfigScreens.xml references regarding ProductConfigOptionList and ListProductConfigItem

modified in product component:
ConfigScreens.xml: from form ref to grid ref , additional cleanup
ConfigForms.xml: from form definition with list ref to grid definition with list ref, additional clean-up

PierreSmits commented:
Hi Jacques,

I have corrected the situation with an additional commit to this PR. I overlooked the references to those two definitions in ConfigScreens.xml.

Your 'Is it normal' threw me off. My answer to the more generic question can only, after more than 13 years of reviewing OFBiz code and practices, and contributing to this project, be that it is normal that things get overlooked by both non- and privileged contributors. 

Thanks again for your feedback and collaboration.
